tests/pipeline/test_computable_term.py

```diff
@@ -0,0 +1,290 @@
+"""Tests for common behaviors shared by all ComputableTerms.
+"""
+import numpy as np
+
+from zipline.lib.labelarray import LabelArray
+from zipline.pipeline import Classifier, Factor, Filter
+from zipline.testing import parameter_space
+from zipline.utils.numpy_utils import (
+    categorical_dtype,
+    datetime64ns_dtype,
+    float64_dtype,
+    int64_dtype,
+    NaTns,
+)
+
+from .base import BaseUSEquityPipelineTestCase
+
+
+class Floats(Factor):
+    inputs = ()
+    window_length = 0
+    dtype = float64_dtype
+
+
+class AltFloats(Factor):
+    inputs = ()
+    window_length = 0
+    dtype = float64_dtype
+
+
+class Dates(Factor):
+    inputs = ()
+    window_length = 0
+    dtype = datetime64ns_dtype
+
+
+class AltDates(Factor):
+    inputs = ()
+    window_length = 0
+    dtype = datetime64ns_dtype
+
+
+class Bools(Filter):
+    inputs = ()
+    window_length = 0
+
+
+class AltBools(Filter):
+    inputs = ()
+    window_length = 0
+
+
+class Strs(Classifier):
+    inputs = ()
+    window_length = 0
+    dtype = categorical_dtype
+    missing_value = None
+
+
+class AltStrs(Classifier):
+    inputs = ()
+    window_length = 0
+    dtype = categorical_dtype
+    missing_value = None
+
+
+class Ints(Classifier):
+    inputs = ()
+    window_length = 0
+    dtype = int64_dtype
+    missing_value = -1
+
+
+class AltInts(Classifier):
+    inputs = ()
+    window_length = 0
+    dtype = int64_dtype
+    missing_value = -1
+
+
+class FillNATestCase(BaseUSEquityPipelineTestCase):
```

Comment:
Are there any tests for `.if_else()` directly?  I know `.fillna` uses `.if_else()` but would it be worth testing for `.if_else()` specifically?

Reply:
I hadn't added these b/c I wasn't sure about the API for `if_then` (in an earlier draft it was `Factor.if_else` rathern thatn `Filter.if_then`. I think I'm pretty happy with this though, so I can add some standalone tests.

Reply:
Added a suite of direct tests for `if_else` here: https://github.com/quantopian/zipline/pull/2691/files#diff-a6c64e76bf199a87f0d8fd09291d8db1R1128.